astropy/visualization/wcsaxes/tests/test_images.py

```diff
@@ -664,14 +665,28 @@ def test_patches_distortion(self, tmpdir):
                    edgecolor='red', facecolor='none')
 
         # World coordinates (should not be distorted)
-        r = SphericalCircle((266.4 * u.deg, -29.1 * u.deg), 0.15 * u.degree,
-                            edgecolor='purple', facecolor='none',
-                            transform=ax.get_transform('fk5'))
-        ax.add_patch(r)
+        r1 = SphericalCircle((266.4 * u.deg, -29.1 * u.deg), 0.15 * u.degree,
+                             edgecolor='purple', facecolor='none',
+                             transform=ax.get_transform('fk5'))
+        ax.add_patch(r1)
+
+        r2 = SphericalCircle(SkyCoord(266.4 * u.deg, -29.1 * u.deg), 0.15 * u.degree,
+                             edgecolor='purple', facecolor='none',
+                             transform=ax.get_transform('fk5'))
+
+        r3 = SphericalCircle(SkyCoord(x=-0.05486461, y=-0.87204803, z=-0.48633538, representation_type='cartesian'),
+                             0.15 * u.degree, edgecolor='purple',
+                             facecolor='none', transform=ax.get_transform('fk5'))
 
         ax.coords[0].set_ticklabel_visible(False)
         ax.coords[1].set_ticklabel_visible(False)
 
+        # Test to verify that SphericalCircle works irrespective of whether
+        # the input(center) is a tuple or a SkyCoord object.
+        assert (r1.get_xy() == r2.get_xy()).all()
+        assert np.allclose(r1.get_xy(), r3.get_xy())
```

Comment:
The test with a `SkyCoord` in Cartesian coordinates raises an `AstropyUserWarning`. You've told `pytest` to ignore this warning on line 632, but an even better approach would be to do the following instead:
```suggestion
        with pytest.warns(AstropyUserWarning, match=expected_message):
            assert np.allclose(r1.get_xy(), r3.get_xy())
```
where `expected_message` is the warning message you expect to receive. That way it becomes possible to test that the code only raises a warning where we expect it to, and the warning type and message are tested too.

Reply:
Ooh, I was looking for a good solution, sorry for not asking. I've made the changes. And now, should I squash all the commits?